cllc-public-app/Controllers/ApplicationsController.cs

```diff
@@ -365,8 +365,23 @@ public IActionResult GetResolvedLGApplications([FromQuery] int pageIndex = 0, [F
                         "adoxio_LicenceFeeInvoice",
                         "adoxio_Invoice"
                     };
+                    var customHeaders = new Dictionary<string, List<string>>();
+                    var preferHeader = new List<string>();
+                    preferHeader.Add($"odata.maxpagesize={pageSize}");
+
+                    customHeaders.Add("Prefer", preferHeader);
+                    var applicationQuery = _dynamicsClient.Applications.GetWithHttpMessagesAsync(filter: filter, expand: expand, customHeaders: customHeaders).GetAwaiter().GetResult();
+
+                    while (pageIndex != 0)
```

Comment:
I would prefer to see this logic changed to:

```csharp
  while (pageIndex > 0) { ... }
```

Since the `pageIndex` param is sent by the client, a knowing party could abuse or trigger a Denial-Of-Service by sending a negative index in the request. The current code would loop forever if the initial pageIndex is negative.

What do you think?

Reply:
done, good catch.